setup.py

```diff
@@ -7,7 +7,12 @@
 
 extras_require = {
     "aws": ["aiobotocore>=0.10.2"],
-    "azure": ["azureml-sdk>=1.0.83"],
+    "azure": [
+        "azureml-sdk>=1.0.83",
```

Comment:
can we add a new one called `azureml` or `aml` to separate out the `azureml-sdk` install? it's really bloated

Reply:
Yeah that might be a good idea.

Reply:
I think to do this neatly we would also need to move the azureml code into its own provider. So it would go from `dask_cloudprovider.azure.AzureMLCluster` to `dask_cloudprovider.azureml.AzureMLCluster`. But `dask_cloudprovider.azure.AzureVMCluster` would remain the same.

Reply:
I'm going to make a PR to remove `AzureMLCluster` after this is in - maybe we should do it as one PR?

Reply:
maybe we should leave it in with deprecation notice

Reply:
> I'm going to make a PR to remove AzureMLCluster after this is in

😂 let's deprecate it for a release or two before removing